.github/workflows/test.yml

```diff
@@ -22,19 +22,31 @@ jobs:
   system-test:
     runs-on: ubuntu-latest
     steps:
+      - uses: actions/checkout@v2
+      
       - name: Extract branch name
+        id: extract_branch
         shell: bash
         run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
-        id: extract_branch
 
-      - uses: actions/checkout@v2
+      - name: Determine the default stage for this run
+        run: echo "STAGE=${{ steps.extract_branch.outputs.branch }}" >> $GITHUB_ENV
+        if: steps.extract_branch.outputs.branch
+
+      - name: Determine the stage for this run if overridden by the workflow_dispatch event
+        run: echo "STAGE=${{ github.event.inputs.stage }}" >> $GITHUB_ENV
+        if: github.event.inputs.stage
+
+      - run: echo "Stage is ${{ env.STAGE }}"  
 
       - name: Run the tests
         uses: ./
         id: test_run
         with:
-          stage: ${{ steps.extract_branch.outputs.branch }}
+          stage: ${{ env.STAGE }} 
 
-      # Use the output from the test run step
-      # - name: Get the output
-        # run: echo "The time was ${{ steps.test_run.outputs.output_1 }}"
\ No newline at end of file
+      # - name: Upload Notebook Output
```

Comment:
I spent some trying trying to get the output notebooks to be written from the Docker container they're created in to the host filesystem and then use upload-artifact to save them, but I couldn't get it to work right. Deferring this until we think it might be usefule.